### ISSUE TITLE
GetHexColor() doesn't wrap index 0 correctly

### ISSUE THREAD

**Issue by shockdude** [dguzek/Simply-Love-SM5#106]
In SL-Colors.lua, GetHexColor(0) returns #FF3C23 (Orange) when it should return #FF7D00 (Yellowish-Orange). This bug is most obvious when selecting #FF003C (Red) as the theme color; the Novice and Easy difficulty colors are the same color when they should be different colors.

edit: corrected wrong color descriptions.

**Comment by dguzek** [comment]
Thanks for finding this issue and filing a bug report.

Work on Simply Love for SM5 is presently halted indefinitely, and I very much hope that it stays that way.  In that, this is unlikely to ever get fixed.  Sorry about this.

**Comment by shockdude** [comment]
Thanks for the work you've done with SL. Been playing Stepmania for a while but only started using SL a few weeks before v4.7. Good luck with your future endeavors.

Anyone interested in fixing this bug themselves can open up `Scripts/SL-Colors.lua` and replace `function GetHexColor( n )` with:
```
function GetHexColor( n )
	local clr = ((n - 1) % #SL.Colors) + 1
	if SL.Colors[clr] then
		return color(SL.Colors[clr])
	end
	
	-- if we were passed nil or a non-integer, return white
	return Color.White
end
```
This fixes several color bugs. In addition to the identical difficulty colors in the Red theme, the Orange now colors things orange instead of red.

Optional optimization/cleanup in `function ColorRGB ( n )` made possible after this fix:
```
function ColorRGB ( n )
	local clr = n + SL.Global.ActiveColorIndex
	return GetHexColor(clr)
end
```

**Comment by dguzek** [comment]
I'll just add this change to the master branch so there's no confusion.

**Comment by dougshell** [comment]
I actually should have just tried again. I did something wrong the first time, the code works as intended. I didn't mean to bother, I just didn't know another way to discuss the issue with 'shockdude' other than via this pull request.

**Comment by dguzek** [comment]
It's okay.

This should be resolved as of 0eb48df30bbfed698d9d15b03f201ca496f51ee5.
